### PR TITLE
Fix live specs

### PR DESCRIPTION
**Pull request by ecoologic**

### CODE REVIEW

File: spec/live/target_spec.rb [hunk 1]
```diff
@@ -12,6 +12,6 @@ def valid_attributes
 
   it_should_be_readable :targets, :create => :true
   it_should_be_creatable
-  it_should_be_updatable :email, "bye@example.com"
+  it_should_be_updatable :active, false
```
Comment (ecoologic) [inline spec/live/target_spec.rb:15]: You can't update the email now.

```
[17] pry(main)> ta.email = 'bye2@example.com'
=> "bye2@example.com"
[18] pry(main)> ta.save
I, [2022-01-17T15:18:29.934505 #55859]  INFO -- : put https://z3n-api-client-rb.zendesk.com/api/v2/targets/1900001126388.json
D, [2022-01-17T15:18:29.934567 #55859] DEBUG -- :   Accept: "application/json"
  Accept-Encoding: "gzip;q=1.0,deflate;q=0.6,identity;q=0.3"
  User-Agent: "ZendeskAPI Ruby 1.34.0"
{:target=>#<ZendeskAPI::SilentMash email="bye2@example.com">}
I, [2022-01-17T15:18:30.607308 #55859]  INFO -- : Status 400 {"error"=>"ParameterMissing", "description"=>"Parameter target is required"}
D, [2022-01-17T15:18:30.607412 #55859] DEBUG -- :   date: "Mon, 17 Jan 2022 05:18:30 GMT"
  content-type: "application/json; charset=UTF-8"
  content-length: "73"
  connection: "keep-alive"
  zendesk-api-version: "2022-01-01"
  x-zendesk-api-version: "v2"
  x-zendesk-application-version: "v10873"
  x-zendesk-api-warn: "Removed Params: Unpermitted parameter keys: '[\"target.email\"]'."
  x-frame-options: "SAMEORIGIN"
  x-rate-limit: "2500"
  x-rate-limit-remaining: "2499"
  strict-transport-security: "max-age=31536000;"
  cache-control: "no-cache"
  x-zendesk-origin-server: "classic-app-server-795b9478dd-vrhtd"
  x-request-id: "6ced1f8fb8306a0e-NRT, 6ced1f8fb8306a0e-NRT"
  x-runtime: "0.151106"
  x-zendesk-zorg: "yes"
  cf-cache-status: "DYNAMIC"
  expect-ct: "max-age=604800, report-uri=\"https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct\""
  report-to: "{\"endpoints\":[{\"url\":\"https:\\/\\/a.nel.cloudflare.com\\/report\\/v3?s=drBMUAy7GODZy%2BA%2Fpr2pXBeg1ZYSHCU7hYl2tsMwoqDRNJ%2BhD2bPp97wudhDPCCJEHC0qJgWgYFL051KVTRhgDA48yFQ7lZaSkv3nFfSskjBPQCDFg7WPFGpg20fWKFW76X2gY8ErnNx8Jxa2kws\"}],\"group\":\"cf-nel\",\"max_age\":604800}"
  nel: "{\"success_fraction\":0.01,\"report_to\":\"cf-nel\",\"max_age\":604800}"
  server: "cloudflare"
  cf-ray: "6ced1f8fb8306a0e-SYD"
{"error"=>"ParameterMissing", "description"=>"Parameter target is required"}
W, [2022-01-17T15:18:30.607536 #55859]  WARN -- : WARNING: Removed Params: Unpermitted parameter keys: '["target.email"]'.
=> false
[19] pry(main)> ta.subject = 'test'
=> "test"
[20] pry(main)> ta.save
I, [2022-01-17T15:19:30.970614 #55859]  INFO -- : put https://z3n-api-client-rb.zendesk.com/api/v2/targets/1900001126388.json
D, [2022-01-17T15:19:30.970706 #55859] DEBUG -- :   Accept: "application/json"
  Accept-Encoding: "gzip;q=1.0,deflate;q=0.6,identity;q=0.3"
  User-Agent: "ZendeskAPI Ruby 1.34.0"
{:target=>#<ZendeskAPI::SilentMash email="bye2@example.com" subject="test">}
I, [2022-01-17T15:19:31.633162 #55859]  INFO -- : Status 400 {"error"=>"ParameterMissing", "description"=>"Parameter target is required"}
D, [2022-01-17T15:19:31.633252 #55859] DEBUG -- :   date: "Mon, 17 Jan 2022 05:19:31 GMT"
  content-type: "application/json; charset=UTF-8"
  content-length: "73"
  connection: "keep-alive"
  zendesk-api-version: "2022-01-01"
  x-zendesk-api-version: "v2"
  x-zendesk-application-version: "v10873"
  x-zendesk-api-warn: "Removed Params: Unpermitted parameter keys: '[\"target.email\", \"target.subject\"]'."
  x-frame-options: "SAMEORIGIN"
  x-rate-limit: "2500"
  x-rate-limit-remaining: "2499"
  strict-transport-security: "max-age=31536000;"
  cache-control: "no-cache"
  x-zendesk-origin-server: "classic-app-server-795b9478dd-5lwr6"
  x-request-id: "6ced210c8be96a09-NRT, 6ced210c8be96a09-NRT"
  x-runtime: "0.166589"
  x-zendesk-zorg: "yes"
  cf-cache-status: "DYNAMIC"
  expect-ct: "max-age=604800, report-uri=\"https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct\""
  report-to: "{\"endpoints\":[{\"url\":\"https:\\/\\/a.nel.cloudflare.com\\/report\\/v3?s=jdTW0sz9OSnPjTqS7xwEgou0NiFvxqi09yHy8g6l6HnpKOcxUb9wgG2%2BFBY%2F4LgWmNYHnpOhjLgVt90kXqY36Z8bmmlvPuKCiAML%2BFyrM1i%2FRzOOj3%2B2BPNV8ZHF%2FeUApO1e0Xge%2Bw6rCh9y5Hak\"}],\"group\":\"cf-nel\",\"max_age\":604800}"
  nel: "{\"success_fraction\":0.01,\"report_to\":\"cf-nel\",\"max_age\":604800}"
  server: "cloudflare"
  cf-ray: "6ced210c8be96a09-SYD"
{"error"=>"ParameterMissing", "description"=>"Parameter target is required"}
W, [2022-01-17T15:19:31.633316 #55859]  WARN -- : WARNING: Removed Params: Unpermitted parameter keys: '["target.email", "target.subject"]'.
=> false
```

File: spec/live/user_spec.rb [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 describe ZendeskAPI::User, :delete_after do
   def valid_attributes
-    { :name => "Test U.", :email => "test@example.org" }
+    { name: "Test U.", email: "test+#{Time.now.to_i}@example.org" }
```
Comment (ecoologic) [inline spec/live/user_spec.rb:5]: I had email uniqueness issues.

```
[31] pry(main)> ZendeskAPI::User.create client, :name => "Test U.", :email => "test@example.org"
I, [2022-01-17T15:58:52.948708 #55859]  INFO -- : post https://z3n-api-client-rb.zendesk.com/api/v2/users
D, [2022-01-17T15:58:52.948777 #55859] DEBUG -- :   Accept: "application/json"
  Accept-Encoding: "gzip;q=1.0,deflate;q=0.6,identity;q=0.3"
  User-Agent: "ZendeskAPI Ruby 1.34.0"
{"user"=>#<ZendeskAPI::SilentMash email="test@example.org" name="Test U.">}
I, [2022-01-17T15:58:53.836941 #55859]  INFO -- : Status 422 {"error"=>"RecordInvalid", "description"=>"Record validation errors", "details"=>{"email"=>[{"description"=>"Email: test@example.org is already being used by another user", "error"=>"DuplicateValue"}]}}
D, [2022-01-17T15:58:53.837055 #55859] DEBUG -- :   date: "Mon, 17 Jan 2022 05:58:53 GMT"
  content-type: "application/json; charset=UTF-8"
  content-length: "193"
  connection: "keep-alive"
  zendesk-api-version: "2022-01-01"
  x-zendesk-api-version: "v2"
  x-zendesk-application-version: "v10873"
  x-frame-options: "SAMEORIGIN"
  x-rate-limit: "2500"
  x-rate-limit-remaining: "2499"
  strict-transport-security: "max-age=31536000;"
  cache-control: "no-cache"
  x-zendesk-origin-server: "classic-app-server-795b9478dd-2wh7g"
  x-request-id: "6ced5ab6885316a9-NRT, 6ced5ab6885316a9-NRT"
  x-runtime: "0.358187"
  x-zendesk-zorg: "yes"
  cf-cache-status: "DYNAMIC"
  expect-ct: "max-age=604800, report-uri=\"https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct\""
  report-to: "{\"endpoints\":[{\"url\":\"https:\\/\\/a.nel.cloudflare.com\\/report\\/v3?s=JH102SocuJEpmxcQl5TCXSVL5Y9Yns2wSc7AG82RvZWonhnrwIPDTgOUU61M9iy4FuwTrLJRUBbCpBcc4f0ZfhVCkjvpp19bb2cO18B6edSRQJgOsqX0nX03iyxrBuoHOSrrt%2BA%2BoxwOJBx5obwy\"}],\"group\":\"cf-nel\",\"max_age\":604800}"
  nel: "{\"success_fraction\":0.01,\"report_to\":\"cf-nel\",\"max_age\":604800}"
  server: "cloudflare"
  cf-ray: "6ced5ab6885316a9-SYD"
{"error"=>"RecordInvalid", "description"=>"Record validation errors", "details"=>{"email"=>[{"description"=>"Email: test@example.org is already being used by another user", "error"=>"DuplicateValue"}]}}
=> nil
```

File: spec/macros/resource_macros.rb [hunk 3]
```diff
@@ -78,6 +80,8 @@ def it_should_be_updatable(attribute, value = "TESTDATA", extra = {})
       end
 
       after(:all) do
+        return unless @object.id
+
```
Comment (ecoologic) [inline spec/macros/resource_macros.rb:84]: These two changes help error readability. The errors in "after test" hide the actual error, the one in creation.

If there's no id, let's not even try to delete the resource.

**Review by tagpro — APPROVED** [review]
ez

**Pull request merged**